### PR TITLE
issue #30729 - Receiving RA serial controlled item, serial from original SO missing from dropdown

### PR DESCRIPTION
**Pull request by jcarlin**
https://github.com/xtuple/xtuple/pull/3171
https://github.com/xtuple/private-extensions/pull/972

### CODE REVIEW

File: common/storedProcErrorLookup.cpp [hunk 1]
```diff
@@ -909,6 +909,9 @@ const struct {
   { "freezeAccountingPeriod", -2, QT_TRANSLATE_NOOP("storedProcErrorLookup", "Cannot freeze this Accounting Period "
 				     "because it is already frozen."), 0, "" },
 
+  { "getAssocLotSerialIds", -1, QT_TRANSLATE_NOOP("storedProcErrorLookup", "pOrdHeadId OR pOrdItemId param is required "
```
Comment (jcarlin) [inline common/storedProcErrorLookup.cpp]: Update this text.

File: guiclient/createLotSerial.cpp [hunk 2]
```diff
@@ -150,20 +116,39 @@ enum SetResponse createLotSerial::set(const ParameterList &pParams)
...
-        lots.bindValue(":itemsite_id", createet.value("itemsite_id").toInt());
-        lots.exec();
-        if (ErrorReporter::error(QtCriticalMsg, this, tr("Error Retrieving Lot/Serial Information"),
-                                      lots, __FILE__, __LINE__))
+      else
+      {
+        // First check for "pre-assigned/associated" lots. A couple example use cases:
+        // 1) Receiving TO - get lot(s) from issue to shipping trans. 2) Receiving RA - get lot(s) from original SO (if exists).
+        XSqlQuery origTransLots;
+        origTransLots.prepare("SELECT ls_number, ls_number FROM ls WHERE ls_id = ANY(getAssocLotSerialIds(pItemlocdistId := :itemlocdist_id));");
```
Comment (gilmoskowitz) [inline guiclient/createLotSerial.cpp]: `IN` would be more clear than and is equivalent to `= ANY`.

File: guiclient/createLotSerial.cpp [hunk 3]
```diff
@@ -150,20 +116,39 @@ enum SetResponse createLotSerial::set(const ParameterList &pParams)
...
+          _lotSerial->setEditable(_privileges->check("EnterUnassociatedLotSerialNumbers"));
+          connect(_lotSerial, SIGNAL(newID(int)), this, SLOT(sLotSerialSelected()));
+        }
+        else if (ErrorReporter::error(QtCriticalMsg, this, tr("Error Retrieving Lot/Serial Information"),
+                                      origTransLots, __FILE__, __LINE__))
         {
           return UndefinedError;
         }
-        else {
+        // No pre-assigned/associated transactions so check if there are lot's in stock to populate menu
```
Comment (gilmoskowitz) [inline guiclient/createLotSerial.cpp]: typo - extra `'`

File: guiclient/createLotSerial.cpp [hunk 4]
```diff
@@ -150,20 +116,39 @@ enum SetResponse createLotSerial::set(const ParameterList &pParams)
...
+        else if (!_serial)
+        {
+          XSqlQuery lots;
+          lots.prepare("SELECT itemloc_id, ls_number FROM itemloc JOIN ls on itemloc_ls_id=ls_id WHERE itemloc_itemsite_id=:itemsite_id;");
+          lots.bindValue(":itemsite_id", _itemsiteid);
+          lots.exec();
+          if (ErrorReporter::error(QtCriticalMsg, this, tr("Error Retrieving Lot/Serial Information"),
```
Comment (gilmoskowitz) [inline guiclient/createLotSerial.cpp]: It's safer to check for errors after calling `_lotSerial->populate()`. Some errors occur while looping through the results.

File: guiclient/enterPoReceipt.cpp [hunk 5]
```diff
@@ -689,16 +689,15 @@ void enterPoReceipt::sPost()
 
     dlg.exec();
   }
-
-  // TODO: update this to sReceiptsUpdated?
-  omfgThis->sPurchaseOrderReceiptsUpdated();
+  else
```
Comment (gilmoskowitz) [inline guiclient/enterPoReceipt.cpp]: This `else` is incorrect.

File: guiclient/createLotSerial.cpp (hunk 2 above)
Comment (jcarlin) [reply guiclient/createLotSerial.cpp]: `IN` isn't an option according to the docs https://www.postgresql.org/docs/9.3/static/arrays.html

**Pull request merged**